### PR TITLE
switched to time interval

### PR DESCRIPTION
**Pull request by derrickstaten**
**Untested / do not merge!**

@shortstuffsushi it should look something like this, if I've done the comparison operation correctly

We can do this in the future, no need to implement today or even this month

Do not thumb up. I just wanted you to see this.

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCSystemObserver.m [hunk 1]
```diff
@@ -131,15 +131,15 @@ + (NSNumber *)getUpdateState {
...
-    int appModificationDay = (int)([[bundleAttributes fileModificationDate] timeIntervalSince1970]/(60*60*24));
-
+    NSDate *modificationDate = [bundleAttributes fileModificationDate];
+    
     if (!storedAppVersion) {
-        if ([documentsDirAttributes fileCreationDate] && [bundleAttributes fileModificationDate] && (appCreationDay != appModificationDay)) {
+        if (creationDate && modificationDate && [modificationDate timeIntervalSinceDate:creationDate] > 60) {
             return [NSNumber numberWithInt:2];
         }
         return nil;
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCSystemObserver.m]: We probably shouldn't ever return `nil` here, right?

File: Branch-SDK/Branch-SDK/BNCSystemObserver.m [hunk 2]
```diff
@@ -131,15 +131,15 @@ + (NSNumber *)getUpdateState {
...
     // for modification date
     NSString *bundleRoot = [[NSBundle mainBundle] bundlePath];
     NSDictionary *bundleAttributes = [manager attributesOfItemAtPath:bundleRoot error:nil];
-    int appModificationDay = (int)([[bundleAttributes fileModificationDate] timeIntervalSince1970]/(60*60*24));
-
+    NSDate *modificationDate = [bundleAttributes fileModificationDate];
+    
     if (!storedAppVersion) {
-        if ([documentsDirAttributes fileCreationDate] && [bundleAttributes fileModificationDate] && (appCreationDay != appModificationDay)) {
+        if (creationDate && modificationDate && [modificationDate timeIntervalSinceDate:creationDate] > 60) {
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCSystemObserver.m]: This comparison would is comparing 60 seconds, is that correct? I thought you may have said as much as minutes.

**Comment by shortstuffsushi** [comment]
No longer untested, and potentially mergeable given additional review (@aaustin).

File: Branch-SDK/Branch-SDK/BNCSystemObserver.m [hunk 3]
```diff
@@ -131,23 +131,34 @@ + (NSNumber *)getUpdateState {
...
+    NSDate *modificationDate = [bundleAttributes fileModificationDate];
+    
+    // No stored version
     if (!storedAppVersion) {
-        if ([documentsDirAttributes fileCreationDate] && [bundleAttributes fileModificationDate] && (appCreationDay != appModificationDay)) {
-            return [NSNumber numberWithInt:2];
+        // Modification and Creation date are more than 60 seconds different indicates an update
+        // This would be the case that they had un-installed and re-installed the app, since this
+        // value isn't currently in NSDefaults.
+        if (creationDate && modificationDate && [modificationDate timeIntervalSinceDate:creationDate] > 60) {
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCSystemObserver.m]: do we need to absolute value this?

**Comment by aaustin** [comment]
Looks great! :+1:

File: Branch-SDK/Branch-SDK/BNCSystemObserver.m (hunk 3 above)
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCSystemObserver.m]: I also considered that, but came to the same conclusion -- the modification date of the bundle couldn't be before the creation date of the the documents directory, right?

**Comment by derrickstaten** [comment]
so abs is really only needed for simulator, but should definitely be in there. love the switch to @1, @2 notation

:+1:

**Comment by derrickstaten** [comment]
if you figure out something with keychain, let's get that in this PR as well

**Comment by shortstuffsushi** [comment]
The keychain research I did indicates that the keychain *is* persisted through uninstall and reinstall. However, we can't grab the creation date of an item in the keychain because we haven't inserted one, in the scenario that this is the first addition of the Branch SDK. We could reevaluate this in the future, but I don't think it's a necessary addition for the time being.

**Pull request merged**